posts/_markdown/2023_quarc.md

```diff
@@ -0,0 +1,40 @@
+---
+title: QUARC
+date: January 23, 2023
+author: Christoph Kirsch and Stefanie Muroya Lei
+---
+
+Bounded Model Checking (BMC) is a well-established technique used to ensure the correctness of critical software. Some examples are verifying properties of software systems and malware detection. However BMC techniques do not generally scale. Thereafter, the aim of our project is to develop a compiler that encodes efficiently classical programs so that quantum computers can help accelerating bounded model checking even in the NISQ-era.
+
+QUARC is a compiler that is included in the [Unicorn toolchain](https://github.com/cksystemsgroup/unicorn). While Unicorn provides a word-level representation and some word-level optimizations such as the integration of powerful SAT and SMT solvers, QUARC takes this word-level representation to generate an oracle for quantum computers so that we can search for inputs that may cause that certain properties of programs are not satisfied. More concretely, QUARC searches for inputs to programs that cause division by zero, invalid memory accesses, or exit codes different than 0.
```

Comment:
```suggestion
**QUARC** is a compiler that is included in the [Unicorn toolchain](https://github.com/cksystemsgroup/unicorn). While Unicorn provides a word-level representation and some word-level optimizations such as the integration of powerful solvers for the Boolean satisfiability problem (SAT) and satisfiability modulo theories (SMT). SMT are a generalization of SAT, and refer to the problem of determining whether a mathematical formula is satisfiable. QUARC takes this word-level representation to generate an oracle for quantum computers so that we can search for inputs that may cause that certain properties of programs are not satisfied. More concretely, QUARC searches for inputs to programs that cause division by zero, invalid memory accesses, or exit codes different than 0.
```